Frontend/src/components/ListingCard.js

```diff
@@ -14,8 +14,7 @@ const ListingCard = ({ setPopupOpen, listingObj }) => {
 
   const handleClick = () => {
     if (!hover) {
-      setPopupOpen(false);
-      history.goBack();
+      handleClose();
```

Comment:
Making this a separate function just makes it harder to read, imo. It's literally only 2 lines of code